tool/release/createnotes/Note.kt

```diff
@@ -0,0 +1,147 @@
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.vaticle.dependencies.tool.release.createnotes
+
+import com.eclipsesource.json.Json
+import com.eclipsesource.json.JsonArray
+import com.eclipsesource.json.JsonObject
+import com.vaticle.dependencies.tool.release.createnotes.Constant.labelBug
+import com.vaticle.dependencies.tool.release.createnotes.Constant.labelFeature
+import com.vaticle.dependencies.tool.release.createnotes.Constant.github
+import com.vaticle.dependencies.tool.release.createnotes.Constant.labelPrefix
+import com.vaticle.dependencies.tool.release.createnotes.Constant.labelRefactor
+
+class Note {
+    enum class Type { FEATURE, BUG, REFACTOR, OTHER }
+
+    private val title: String
+    private val goal: String?
+    private val type: Type
+
+    private constructor(title: String, goal: String?, type: Type) {
+        this.title = title
+        this.goal = goal
+        this.type = type
+    }
+
+    fun toMarkdown():  String {
+        return "- **$title**\n  ${goal ?: ""}"
+    }
+
+    companion object {
+        fun fromGithubPR(pr: JsonObject): Note {
+            return Note(
+                title = pr.get("title").asString(),
+                goal = getPRGoal(pr.get("body").asString()),
+                type = getPRType(pr.get("labels").asArray())
+            )
+        }
+
+        private fun getPRType(labels: JsonArray): Type {
+            val types = labels
+                .map { e -> e.asObject().get("name").asString() }
+                .filter { e -> e.startsWith(labelPrefix) }
+            val type =
+                when {
+                    types.contains(labelFeature) -> Type.FEATURE
+                    types.contains(labelBug) -> Type.BUG
+                    types.contains(labelRefactor) -> Type.REFACTOR
+                    else -> Type.OTHER
+                }
+            return type
+        }
+
+        private fun getPRGoal(description: String): String {
+            val goal = StringBuilder()
+            var header = 0
+            for (line in description.lines()) {
+                if (line.startsWith("##")) {
+                    header += 1
+                } else if (header == 1) {
+                    goal.append(line)
+                } else if (header > 1) {
+                    break
+                }
+            }
+            return goal.toString()
+        }
+
+        fun fromGithubCommit(commit: JsonObject): Note {
+            return Note(commit.get("message").asString().lines().first(), null, Type.OTHER)
+        }
+
+        fun toMarkdown(notes: List<Note>): String {
```

Comment:
collect all notes and render it into a release note, placing each note into features, bugs, refactors, or others section in the release note.